lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -239,17 +631,37 @@ export default function getAPIImplementation(): APIInterface {
             let email = formFields.filter((f) => f.id === "email")[0].value;
             let password = formFields.filter((f) => f.id === "password")[0].value;
 
-            let response = await options.recipeImplementation.signUp({ email, password, userContext });
+            let isSignUpAllowed = await AccountLinking.getInstanceOrThrowError().isSignUpAllowed({
+                newUser: {
+                    recipeId: "emailpassword",
+                    email,
+                },
+                userContext,
+            });
+
+            if (!isSignUpAllowed) {
+                return {
+                    status: "SIGNUP_NOT_ALLOWED",
+                    reason: "the sign-up info is already associated with another account where it is not verified",
+                };
+            }
+            let response = await options.recipeImplementation.signUp({
+                email,
+                password,
+                doAccountLinking: true,
```

Comment:
to discuss this - we should call shouldDoAccountLinking instead of assuming that it's true - or remove this boolean if we are calling shouldDoAccountLinking inside this function already